packages/compass-aggregations/src/components/stage-editor/stage-editor.tsx

```diff
@@ -0,0 +1,239 @@
+import {
+  css,
+  cx,
+  spacing,
+  palette,
+  Banner,
+  withDarkMode,
+} from '@mongodb-js/compass-components';
+import { changeStageValue } from '../../modules/pipeline-builder/stage-editor';
+import { mapPipelineModeToEditorViewType } from '../../modules/pipeline-builder/builder-helpers';
+import type { RootState } from '../../modules';
+import type { PipelineParserError } from '../../modules/pipeline-builder/pipeline-parser/utils';
+
+const { track } = createLoggerAndTelemetry('COMPASS-AGGREGATIONS-UI');
+
+const editorContainerStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+  justifyContent: 'space-between',
+  alignItems: 'stretch',
+  overflow: 'hidden',
+  height: '100%',
+});
+
+const editorStyles = css({
+  flex: 1,
+  flexShrink: 0,
+  margin: 0,
+  width: '100%',
+  minHeight: '230px',
+});
+
+const editorContainerStylesDark = css({
+  background: palette.gray.dark3,
+});
+
+const editorContainerStylesLight = css({
+  background: palette.gray.light3,
+});
+
+const aceEditorStyles = css({
+  minHeight: '160px',
+});
+
+const bannerStyles = css({
+  flex: 'none',
+  marginTop: spacing[2],
+  marginLeft: spacing[2],
+  marginRight: spacing[2],
+  textAlign: 'left',
+});
+
+type StageEditorProps = {
+  darkMode?: boolean;
+  index: number;
+  stageOperator: string | null;
+  stageValue: string | null;
+  serverVersion: string;
+  autocompleteFields: CompletionWithServerInfo[];
+  syntaxError: PipelineParserError | null;
+  serverError: MongoServerError | null;
+  num_stages: number;
+  editor_view_type: string;
+  className: string;
+  onChange: (index: number, value: string) => void;
+};
+
+
+function useStageCompleter(
+  ...args: ConstructorParameters<typeof StageAutoCompleter>
+): StageAutoCompleter {
+  const [version, textCompleter, fields, operator] = args;
+  const completer = useRef<StageAutoCompleter>();
+  if (!completer.current) {
+    completer.current = new StageAutoCompleter(
+      version,
+      textCompleter,
+      fields,
+      operator
+    );
+  }
+  useEffect(() => {
+    completer.current?.update(fields, operator ?? null, version);
+  }, [fields]);
+  return completer.current;
+}
+
+const UnthemedStageEditor = ({
+  stageValue,
+  stageOperator,
+  index,
+  onChange,
+  serverError,
+  syntaxError,
+  darkMode,
+  className,
+  autocompleteFields,
+  serverVersion,
+  num_stages,
+  editor_view_type,
+}: StageEditorProps) => {
+  const editorInitialValueRef = useRef<string | null>(stageValue);
+  const editorRef = useRef<AceEditor | undefined>(undefined);
+  const completer = useStageCompleter(
+    serverVersion,
+    EditorTextCompleter,
+    autocompleteFields,
+    stageOperator
+  );
+
+  // When the index changes and the value is different from the editor value,
+  // update the editor value.
+  useEffect(() => {
+    if (editorRef.current?.getValue() !== stageValue) {
+      editorRef.current?.setValue(stageValue ?? '');
+    }
+  }, [index]);
```

Comment:
While this resolves the issue, this doesn't look like an appropriate fix as the issue indicates that either we are not handling updates correctly in the Editor component, passing the value incorrectly, or using StageEditor incorrectly in focus mode as the issue doesn't occur when you are adding a stage in a stage-by-stage view. We should find the root cause for this